src/binary-writer.cc

```diff
@@ -190,6 +191,117 @@ class Symbol {
   }
 };
 
+class SymbolTable {
+  WABT_DISALLOW_COPY_AND_ASSIGN(SymbolTable);
+
+  std::vector<Symbol> symbols_;
+
+  std::vector<Index> functions_;
+  std::vector<Index> globals_;
+
+  std::set<string_view> seen_names_;
+
+  Result Intern(const string_view& name) {
```

Comment:
Why not just call this `Add`?  or `AddName`?

Reply:
Heh, apologies for the name ;)  I come from the lisp world, which is what this operation is called there... But I can update it to something less jargon-ful.